src/vsr/replica_test.zig

```diff
@@ -1297,15 +1297,18 @@ test "Cluster: upgrade: R=1" {
 }
 
 test "Cluster: upgrade: state-sync to new release" {
-    const t = try TestContext.init(.{ .replica_count = 3, .seed = 2 });
+    const t = try TestContext.init(.{ .replica_count = 3 });
     defer t.deinit();
 
     var c = t.clients(0, t.cluster.clients.len);
 
     t.replica(.R_).stop();
     try t.replica(.R0).open_upgrade(&[_]u8{ 1, 2 });
     try t.replica(.R1).open_upgrade(&[_]u8{ 1, 2 });
-    try c.request(checkpoint_2_trigger, checkpoint_2_trigger);
+    t.run();
+    try expectEqual(t.replica(.R0).commit(), checkpoint_1_trigger);
+    try c.request(constants.vsr_checkpoint_interval, constants.vsr_checkpoint_interval);
```

Comment:
I am wondering whether we should change the semantic of `request` to "advancing to op n" rather than "doing n requests"?

My gut feeling is yes, but curious what you'd think here.

Reply:
:thinking: Overall, I think that makes sense -- that is, it fits how we actually write tests.

The original idea of `var c = t.clients(0, t.cluster.clients.len);` was that there could be multiple "pools" of clients which have their requests managed independently. This approach makes more sense with "do n requests" rather than "advance to op n"... But we don't ever use multiple pools right now. (Though maybe we will soon for eviction tests? But probably not worth designing our API around them.)